yarn-project/noir-contracts/src/scripts/copy_output.ts

```diff
@@ -69,64 +75,67 @@ function getFunction(type: FunctionType, params: ABIParameter[], returns: ABITyp
-        );
-      return getFunction(
-        thisFunctionAbisComments.type || (fn.function_type.toLowerCase() as FunctionType),
-        thisFunctionAbisComments.params || fn.abi.parameters,
-        thisFunctionAbisComments.return || [fn.abi.return_type],
-        fn,
-      );
+      return getFunction(fn.function_type.toLowerCase() as FunctionType, fn.abi.parameters, fn.abi.return_type, fn);
     });
 }
 
 const main = () => {
   const name = process.argv[2];
   if (!name) throw new Error(`Missing argument contract name`);
 
-  const folderName = `${snakeCase(name)}_contract`;
-  const folderPath = `src/contracts/${folderName}`;
-  const source = readFileSync(`${folderPath}/src/main.nr`).toString();
-  const contractName = process.argv[3] ?? upperFirst(camelCase(name));
-  const build = JSON.parse(readFileSync(`${folderPath}/target/${folderName}-${contractName}.json`).toString());
-  const artifacts = 'src/artifacts';
+  const projectName = `${snakeCase(name)}_contract`;
+  const projectDirPath = `src/contracts/${projectName}`;
+  const sourceCodeFilePath = `${projectDirPath}/src/main.nr`;
+  const sourceCode = readFileSync(sourceCodeFilePath).toString();
 
-  const abi = {
-    name: build.name,
-    functions: getFunctions(source, build),
-  };
+  const contractName = upperFirst(camelCase(name));
+  const buildJsonFilePath = `${projectDirPath}/target/${projectName}-${contractName}.json`;
+  const buildJson = JSON.parse(readFileSync(buildJsonFilePath).toString());
 
-  const exampleFile = `${artifacts}/${snakeCase(name)}_contract.json`;
-  writeFileSync(exampleFile, JSON.stringify(abi, null, 2) + '\n');
-  log(`Written ${exampleFile}`);
-
-  writeToProject(abi);
+  // Remove extraneous information from the buildJson (which was output by Nargo) to hone in on the function data we actually care about:
+  const artifactJson: ContractAbi = {
+    name: buildJson.name,
+    functions: getFunctions(sourceCode, buildJson),
+  };
 
-  const typeFile = `src/types/${name}.ts`;
-  writeFileSync(typeFile, generateType(abi, '../artifacts/index.js'));
-  log(`Written ${typeFile}`);
+  // Write the artifact:
+  const artifactsDir = 'src/artifacts';
+  const artifactDestFilePath = `${artifactsDir}/${snakeCase(name)}_contract.json`;
+  writeFileSync(artifactDestFilePath, JSON.stringify(artifactJson, null, 2) + '\n');
+  log(`Written ${artifactDestFilePath}`);
+
+  // Write some artifacts to other packages in the monorepo:
+  writeToProject(artifactJson);
+
+  // Write a .ts contract interface, for consumption by the typescript code
+  const tsInterfaceDestFilePath = `src/types/${name}.ts`;
+  writeFileSync(tsInterfaceDestFilePath, generateTSContractInterface(artifactJson, '../artifacts/index.js'));
+  log(`Written ${tsInterfaceDestFilePath}`);
+
+  // Write a .nr contract interface, for consumption by other Noir Contracts
+  if (INTERFACE_CONTRACTS.includes(name)) {
+    const noirInterfaceDestFilePath = `${projectDirPath}/src/${projectName}_interface.nr`;
```

Comment:
A possible problem with writing the interface here is that this dir is considered a 'binary' dir by nargo (because the Nargo.toml of a contract will have a 'bin' type). That means other nargo packages (I.e. other contracts) won't be able to consume the interface from this directory. 
I guess projects in Solidity land need to copy the interface file over into their own projects, so this is fine?

Reply:
Looks like you're using a symlink to import the test example.

Reply:
>I guess projects in Solidity land need to copy the interface file over into their own projects, so this is fine?

No, Solidity doesn't have this bin/lib limitation that Noir has, so you can freely import anything from any package. I'm not sure how to solve this properly, and was thinking in tackling it as part of #1457. Symlink seemed a reasonable solution for now.

Reply:
Yep, seems reasonable to me, for now. Thanks